src/RpcServer/RpcServer.Wallet.cs

```diff
@@ -97,6 +114,16 @@ private JObject ImportPrivKey(JArray _params)
             };
         }
 
+        [RpcMethod]
+        private JObject CalculateNetworkFee(JArray _params)
+        {
+            byte[] tx = Convert.FromBase64String(_params[0].AsString());
+
+            JObject account = new JObject();
+            account["networkfee"] = (wallet ?? new DummyWallet()).CalculateNetworkFee(Blockchain.Singleton.GetSnapshot(), tx.AsSerializable<Transaction>());
```

Comment:
This requires the user to open wallet? then rpcclient need to call openwallet first to get the right networkfee.

Reply:
no, if it's not open, it will create a dummy wallet

Reply:
then the networkfee is incorrect, and the tx will be invalid.

Reply:
Cand you detail the error?

Reply:
https://github.com/neo-project/neo/blob/4066a4f3986a71612dcdeca33a4ecf8934511247/src/neo/Wallets/Wallet.cs#L373-L374

`GetAccount(hash)` returns empty.

Reply:
mmm... then is not a good solution, because you need to use the adresses of the rpc server...

Reply:
Maybe we should rename RpcClient to CSharpSDK, then we can open wallet and create transaction and relay it.

Reply:
This way you can use Neo.dll directly.

Reply:
I think that the unique way is create the witnesses using the client wallet, and send them in the transaction, then with https://github.com/neo-project/neo/pull/1931 it could be easily computed.